metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -1,20 +1,174 @@
+import org.hypertrace.core.query.service.client.QueryServiceClient;
+import org.hypertrace.core.query.service.client.QueryServiceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class MetricAnomalyDetector extends PlatformService {
+class MetricAnomalyDetector {
 
-  @Override
-  protected void doInit() {}
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetricAnomalyDetector.class);
 
-  @Override
-  protected void doStart() {}
+  private static final String QUERY_SERVICE_CONFIG_KEY = "query.service.config";
+  private static final String REQUEST_TIMEOUT_CONFIG_KEY = "request.timeout";
+  private static final int DEFAULT_REQUEST_TIMEOUT_MILLIS = 10000;
+  private static final String METRIC_ANOMALY_EVENT_CONDITION = "MetricAnomalyEventCondition";
 
-  @Override
-  protected void doStop() {}
+  static final String TENANT_ID_KEY = "x-tenant-id";
 
-  @Override
-  public boolean healthCheck() {
-    return false;
+  private final MetricQueryBuilder metricQueryBuilder;
+  private final QueryServiceClient queryServiceClient;
+  private final int qsRequestTimeout;
+
+  MetricAnomalyDetector(Config appConfig) {
+    AttributeServiceClientConfig asConfig = AttributeServiceClientConfig.from(appConfig);
+    ManagedChannel attributeServiceChannel =
+        ManagedChannelBuilder.forAddress(asConfig.getHost(), asConfig.getPort())
+            .usePlaintext()
+            .build();
+    AttributeServiceClient asClient = new AttributeServiceClient(attributeServiceChannel);
+
+    Config qsConfig = appConfig.getConfig(QUERY_SERVICE_CONFIG_KEY);
+    queryServiceClient = new QueryServiceClient(new QueryServiceConfig(qsConfig));
+    qsRequestTimeout =
+        appConfig.hasPath(REQUEST_TIMEOUT_CONFIG_KEY)
+            ? appConfig.getInt(REQUEST_TIMEOUT_CONFIG_KEY)
+            : DEFAULT_REQUEST_TIMEOUT_MILLIS;
+
+    metricQueryBuilder = new MetricQueryBuilder(asClient);
+  }
+
+  void process(AlertTask alertTask) {
+    MetricAnomalyEventCondition metricAnomalyEventCondition;
+    if (alertTask.getEventConditionType().equals(METRIC_ANOMALY_EVENT_CONDITION)) {
+      try {
+        metricAnomalyEventCondition =
+            MetricAnomalyEventCondition.parseFrom(alertTask.getEventConditionValue());
+      } catch (InvalidProtocolBufferException e) {
+        LOGGER.error("Exception while parsing event condition", e);
+        return;
+      }
+    } else {
+      LOGGER.debug("EventConditionType:{}", alertTask.getEventConditionType());
+      return;
+    }
+
+    QueryRequest queryRequest =
+        metricQueryBuilder.buildMetricQueryRequest(
+            metricAnomalyEventCondition.getMetricSelection(),
+            alertTask.getLastExecutionTime(),
+            alertTask.getCurrentExecutionTime(),
+            alertTask.getTenantId());
+
+    // todo handle multiple violation conditions
+    ViolationCondition violationCondition =
+        metricAnomalyEventCondition.getViolationConditionList().get(0);
```

Comment:
Do we have to check if its empty - metricAnomalyEventCondition.getViolationConditionList()?

Reply:
I didn't check assuming that the producer will filter out such rules

Reply:
Currently, it's not doing that. Can we add that condition as I saw some other conditions too?